common/thorhelper/thorcommon.cpp

```diff
@@ -1823,14 +1823,26 @@ void setAutoAffinity(unsigned curProcess, unsigned processPerMachine, const char
     unsigned numaMap[NUMA_NUM_NODES];
     unsigned numNumaNodes = 0;
 #if defined(LIBNUMA_API_VERSION) && (LIBNUMA_API_VERSION>=2)
-    for (unsigned i=0; i<=numa_max_node(); i++)
+    unsigned maxcpus = numa_num_configured_cpus();
+    for (unsigned cpu=0; cpu < maxcpus; cpu++)
     {
-        if (numa_bitmask_isbitset(numa_all_nodes_ptr, i))
+        int node = numa_node_of_cpu(cpu);
+        if (node != -1)
         {
-            numaMap[numNumaNodes] = i;
-            numNumaNodes++;
+            unsigned cur;
+            for (cur = 0; cur < numNumaNodes; cur++)
+            {
+                if (numaMap[cur] == node)
+                    break;
+            }
+            if (cur == numNumaNodes)
+            {
+                numaMap[numNumaNodes] = node;
+                numNumaNodes++;
+            }
         }
     }
+    DBGLOG("Affinity: Max cpus(%u) nodes(%u) actual nodes(%u)", maxcpus, numa_max_node(), numNumaNodes);
```

Comment:
minor - numa_max_node() returns the highest node number, not the number of nodes.
numa_num_configured_nodes() might provide number of nodes irregardless of if they have memory or not.